operator/controllers/btpoperator_controller.go

```diff
@@ -199,39 +404,63 @@ func (r *BtpOperatorReconciler) HandleInitialState(ctx context.Context, cr *v1al
 	return r.UpdateBtpOperatorStatus(ctx, cr, types.StateProcessing, Initialized, "Initialized")
 }
 
+func (r *BtpOperatorReconciler) CheckIfReasonIsSet(cr *v1alpha1.BtpOperator, reason Reason) bool {
```

Comment:
you can create method for `*v1alpha1.BtpOperator` to check the Reasons and then call `cr.isReasonEqual`